packages/client/src/devices/InputMediaDeviceManagerState.ts

```diff
@@ -1,16 +1,20 @@
-import { BehaviorSubject, distinctUntilChanged, Observable } from 'rxjs';
+import { BehaviorSubject, distinctUntilChanged } from 'rxjs';
 import { RxUtils } from '../store';
 
 export type InputDeviceStatus = 'enabled' | 'disabled' | undefined;
 
-export abstract class InputMediaDeviceManagerState {
+export abstract class InputMediaDeviceManagerState<C = MediaTrackConstraints> {
   protected statusSubject = new BehaviorSubject<InputDeviceStatus>(undefined);
   protected mediaStreamSubject = new BehaviorSubject<MediaStream | undefined>(
     undefined,
   );
   protected selectedDeviceSubject = new BehaviorSubject<string | undefined>(
     undefined,
   );
+  protected defaultConstraintsSubject = new BehaviorSubject<C | undefined>(
```

Comment:
What's the aim of this object? As far as I can tell, it's always undefined

Reply:
It should allow easier override of the default constraints. Example:

```
call.microphone.setDefaultConstraints({
  echoCancellation: true,
})
```

One method is missing in this implementation, I'll add it now: `InputMediaDeviceManager#setDefaultConstraints`.

Reply:
I see, it makes sense now, thanks